core/src/components/tab-button/tab-button.tsx

```diff
@@ -89,9 +89,20 @@ export class TabButton implements ComponentInterface {
     return !!this.el.querySelector('ion-icon');
   }
 
+  private get tabIndex() {
+    const hasTabIndex = this.el.hasAttribute('tabindex');
+
+    if (hasTabIndex) {
+      return this.el.getAttribute('tabindex');
+    }
+
+    return 0;
```

Comment:
Should it still return 0 if the tab button is disabled? I don't believe you should be able to tab to disabled elements: https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XUL/Attribute/disabled